### PR TITLE
Refactor/change password

### LINKED ISSUE #411: Switch to new AdminSetUserPassword API
Just an FYI as someone who's been waiting for this feature for a while I noticed that Cognito finally released an API that gives you the ability to change a user's password for them.

https://aws.amazon.com/about-aws/whats-new/2019/05/amazon-cognito-launches-enhanced-user-password-reset-api-for-administrators/

Hopefully, this will help address one of the major limitations.  If I have time later maybe I'll submit my own PR.

### PR DESCRIPTION
**Pull request by rafaelsiqueira**
Closes #411

### CODE REVIEW

**Comment by rafaelsiqueira** [comment]
@squidfunk recommended review by commits.

File: share/lambda/Makefile [hunk 1]
```diff
@@ -61,7 +61,7 @@ dist.zip: $(shell find src) | dist-pre dist dist-post
 		&& cd dist \
 		&& cp ../package*.json . \
 		&& npm install --production \
-		&& rm -rf package*.json node_modules/aws-sdk etc \
```
Comment (squidfunk) [inline share/lambda/Makefile:64]: What's the reason for removal? `aws-sdk` is already available in Lambda.

**Review by squidfunk — COMMENTED** [review]
This PR is awesome! Thanks for providing it. Questions:

1. See the comment on bundling `aws-sdk`
2. Did you run the integration tests? I'm a little tight on time and can merge and release this when we're sure it works, but I have little time to test it myself.

File: share/lambda/Makefile (hunk 1 above)
Comment (rafaelsiqueira) [reply share/lambda/Makefile:64]: To be honest, it was already on the original branch. I'll put it back and run the tests.

Comment (rafaelsiqueira) [reply share/lambda/Makefile]: Done.

**Comment by rafaelsiqueira** [comment]
> This PR is awesome! Thanks for providing it. Questions:
> 1. See the comment on bundling `aws-sdk`
Done.
> 2. Did you run the integration tests? I'm a little tight on time and can merge and release this when we're sure it works, but I have little time to test it myself.
Done. All tests passing.

File: modules/api/lambda/package.json [hunk 2]
```diff
@@ -23,7 +23,6 @@
     "lint": "make lint",
     "start": "make start",
     "test": "make test",
-    "test:acceptance": "make test-acceptance",
```
Comment (squidfunk) [inline modules/api/lambda/package.json:26]: Why did you remove the acceptance tests script?

**Review by squidfunk — COMMENTED** [review]
I skimmed through the PR again, just one more finding, see comment. Otherwise, it's ready to merge. Thanks again for your work!

File: modules/api/lambda/package.json (hunk 2 above)
Comment (rafaelsiqueira) [reply modules/api/lambda/package.json:26]: I didn’t. Original branch again. Will rollback this too.

Comment (squidfunk) [reply modules/api/lambda/package.json:26]: Oh my, sorry for the confusion. I haven't been working on this project for quite some time now. All good. It's fine.
